pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -260,19 +268,32 @@ public HelixAdmin getHelixAdmin() {
    */
   private HelixManager registerAndConnectAsHelixParticipant() {
     HelixManager helixManager = HelixManagerFactory
-        .getZKHelixManager(_helixClusterName, CommonConstants.Helix.PREFIX_OF_CONTROLLER_INSTANCE + _instanceId,
-            InstanceType.PARTICIPANT, _helixZkURL);
+        .getZKHelixManager(_helixClusterName, _controllerParticipantInstanceId, InstanceType.PARTICIPANT, _helixZkURL);
+
+    StateMachineEngine stateMach = helixManager.getStateMachineEngine();
```

Comment:
Line 273-275 can be merged into one line, and please add some comments for this.

Reply:
Done.